docs/dsl1.rst

```diff
@@ -0,0 +1,192 @@
+        .map{ it.reverse() }
+        .view()
+
+Similarly, process outputs can be consumed by multiple consumers automatically, which makes workflow scripts
+much easier to read and write.
+
+
+Modules
+=======
+
+In DSL1, the entire Nextflow pipeline must be defined in a single file (e.g. ``main.nf``). This
+restriction becomes quite cumbersome as a pipeline becomes larger, and it hinders the sharing and
+reuse of pipeline components.
+
+DSL2 introduces the concept of "module scripts" (or "modules" for short), which are Nextflow scripts
+that can be "included" by other scripts. While modules are not essential to migrating to DSL2, nor are
+they mandatory in DSL2 by any means, modules can help you organize a large pipeline into multiple smaller
+files, and take advantage of modules created by others. Check out the :ref:`module-page` to get started.
+
+
+Deprecations
+============
+
+Processes
+---------
+
+* The ``set`` process input type is no longer supported, use :ref:`tuple <process-input-tuple>` instead.
+* The ``set`` process output type is no longer supported, use :ref:`tuple <process-out-tuple>` instead.
+* The ``mode flatten`` option for process outputs is no longer available. Use the :ref:`operator-flatten` operator on the corresponding output channel instead.
+
+* Unqualified value and file elements in a tuple declaration are no longer allowed. Use an explicit
+  ``val`` or ``path`` qualifier.
+  
+  For example::
+
+    process foo {
+        input:
+        tuple X, 'some-file.sam'
+        output:
+        tuple X, 'some-file.bam'
+
+        script:
+        '''
+        your_command --in $X some-file.sam > some-file.bam
+        '''
+    }
+
+  Use::
+
+    process foo {
+        input:
+        tuple val(X), path('some-file.sam')
+        output:
+        tuple val(X), path('some-file.bam')
+
+        script:
+        '''
+        your_command --in $X some-file.sam > some-file.bam
+        '''
+    }
+
+
+Channels
+--------
+
+* Channel method ``bind`` has been deprecated in DSL2.
+* Channel method ``<<`` has been deprecated in DSL2.
+* Channel factory ``create`` has been deprecated in DSL2.
+
+Operators
+---------
+
+* Operator ``choice`` has been deprecated in DSL2. Use :ref:`operator-branch` instead.
+* Operator ``close`` has been deprecated in DSL2.
+* Operator ``countBy`` has been deprecated in DSL2.
+* Operator ``into`` has been deprecated in DSL2, as it is no longer needed.
+* Operator ``fork`` has been renamed to :ref:`operator-multimap`.
+* Operator ``groupBy`` has been deprecated in DSL2. Use :ref:`operator-grouptuple` instead.
+* Operators ``print`` and ``println`` have been deprecated in DSL2. Use :ref:`operator-view` instead.
+* Operator ``route`` has been deprecated in DSL2.
+* Operator ``separate`` has been deprecated in DSL2.
+* Operator ``spread`` has been deprecated in DSL2. Use :ref:`operator-combine` instead.
+
+DSL2 Preview
+------------
+
+* The ``nextflow.preview.dsl=2`` feature flag is no longer needed.
```

Comment:
```suggestion
An early preview of DSL2 was available in 2020. Note that some of that early DSL2 syntax has since changed.

* The ``nextflow.preview.dsl=2`` (and ``nextflow.enable.dsl=1``) feature flags are no longer needed.
```